ui/push-health/Health.jsx

```diff
@@ -87,6 +91,50 @@ export default class Health extends React.PureComponent {
     this.setState({ user });
   };
 
+  setRegressionsOrderBy = (regressionsOrderBy) => {
+    const { location, history } = this.props;
+    const newParams = {
+      ...parseQueryParams(location.search),
+      regressionsOrderBy,
+    };
+    const queryString = createQueryParams(newParams);
+    updateQueryParams(queryString, history, location);
+    this.setState({ regressionsOrderBy });
+  };
+
+  setRegressionsGroupBy = (regressionsGroupBy) => {
+    const { location, history } = this.props;
+    const newParams = {
+      ...parseQueryParams(location.search),
+      regressionsGroupBy,
+    };
+    const queryString = createQueryParams(newParams);
+    updateQueryParams(queryString, history, location);
+    this.setState({ regressionsGroupBy });
+  };
+
+  setKnownIssuesOrderBy = (knownIssuesOrderBy) => {
+    const { location, history } = this.props;
+    const newParams = {
+      ...parseQueryParams(location.search),
+      knownIssuesOrderBy,
+    };
+    const queryString = createQueryParams(newParams);
+    updateQueryParams(queryString, history, location);
+    this.setState({ knownIssuesOrderBy });
+  };
+
+  setKnownIssuesGroupBy = (knownIssuesGroupBy) => {
+    const { location, history } = this.props;
+    const newParams = {
+      ...parseQueryParams(location.search),
+      knownIssuesGroupBy,
+    };
+    const queryString = createQueryParams(newParams);
+    updateQueryParams(queryString, history, location);
+    this.setState({ knownIssuesGroupBy });
+  };
```

Comment:
You're on the right track here, but since all of this logic is the same you should create one generic function that accepts an argument for the state that should be updated (there is a way to do this without entering a specific state name/key). I suggest naming it something like `updateParamsAndState`.

Reply:
I was facing the problem of updating the state variable from a generic function
I mean, how do I come to know which state variable ```regressionsGroupBy``` or ```knownIssuesOrderBy``` is to be changed
Other things will be surely much better to handle in a generic function

Reply:
There is a way to solve this problem, you just need to research it :)

Reply:
It is better to find a DRY solution - here you are repeating the same code multiple times. So there are a few different approaches you can take that would be better.

Reply:
I am unable to find a good solution to it.
The only way I find is to pass a string parameter from the classification group component to see which state variable has to be changed. Does this method work?

Reply:
An easy solution is to pass an object parameter instead of a variable: `this.setState(stateObj)`. Another one would be to call setState inline in the onClick and update the query parameters as a callback function.

Reply:
You could pass in a string, but then you'd need a value unless you were doing a boolean toogle, so: `(stateName, value) => this.setState({ [stateName]: value }). This is possible by using computed property names.